packages/ember-runtime/lib/computed/reduce_computed_macros.js

```diff
@@ -554,31 +554,38 @@ export function sort(itemsKey, sortDefinition) {
   }
 }
 
+function remove(collection, stuffToRemove) {
+  for (var i = collection.length - 1; i >= 0; i--) {
+    for (var j = stuffToRemove.length - 1; j >= 0; j--) {
+      if (collection[i] === stuffToRemove[j]) {
+        collection.splice(i, 1);
+        stuffToRemove.splice(j, 1);
+      }
+    }
+  }
+}
+
 function customSort(itemsKey, comparator) {
   return arrayMacro(itemsKey, function(value) {
     return value.slice().sort((x, y) => comparator.call(this, x, y));
   });
 }
-
 // This one needs to dynamically set up and tear down observers on the itemsKey
 // depending on the sortProperties
 function propertySort(itemsKey, sortPropertiesKey) {
-  var cp = new ComputedProperty(function(key) {
-    function didChange() {
-      this.notifyPropertyChange(key);
-    }
+  var sym = symbol('sortDependentKeys');
 
+  var cp = new ComputedProperty(function(key) {
     var items = itemsKey === '@this' ? this : get(this, itemsKey);
     var sortProperties = get(this, sortPropertiesKey);
 
     if (items === null || typeof items !== 'object') { return Ember.A(); }
 
-    // TODO: Ideally we'd only do this if things have changed
-    if (cp._sortPropObservers) {
-      cp._sortPropObservers.forEach(args => removeObserver.apply(null, args));
-    }
+    var sortDependentKeys = cp[sym];
 
-    cp._sortPropObservers = [];
+    if (sortDependentKeys) {
+      remove(cp._dependentKeys, sortDependentKeys);
```

Comment:
can this be an method of CP?  I'm hoping to reduce bookkeeping meta stuff from being sprinkled around thing, also CP descs are shared between instances, is this sym specific to this instance?